pkg/routeagent_driver/handlers/ovn/subnets.go

```diff
@@ -18,15 +18,20 @@ limitations under the License.
 
 package ovn
 
-import "k8s.io/utils/set"
+import (
+	k8snet "k8s.io/utils/net"
+	"k8s.io/utils/set"
+)
 
 func (ovn *Handler) getRemoteSubnets() set.Set[string] {
 	endpointSubnets := set.New[string]()
 
 	endpoints := ovn.State().GetRemoteEndpoints()
 	for i := range endpoints {
 		for _, subnet := range endpoints[i].Spec.Subnets {
-			endpointSubnets.Insert(subnet)
+			if k8snet.IPFamilyOfCIDRString(subnet) == ovn.ipFamily {
+				endpointSubnets.Insert(subnet)
+			}
```

Comment:
We could actually replace L31 - L35 with:

`endpointSubnets.Insert(cidr.ExtractSubnets(ovn.ipFamily, endpoints[i].Spec.Subnets)...)`